src/Symfony/Component/HttpKernel/Config/FileLocator.php

```diff
@@ -22,18 +22,28 @@
 class FileLocator extends BaseFileLocator
 {
     private $kernel;
-    private $path;
 
     /**
-     * @param string|null $path  The path the global resource directory
-     * @param array       $paths An array of paths where to look for resources
+     * @deprecated since Symfony 4.4
      */
-    public function __construct(KernelInterface $kernel, string $path = null, array $paths = [])
+    private $path;
+
+    public function __construct(KernelInterface $kernel/*, string $path = null, array $paths = [], bool $triggerDeprecation = true*/)
     {
         $this->kernel = $kernel;
-        if (null !== $path) {
-            $this->path = $path;
-            $paths[] = $path;
+
+        if (2 <= \func_num_args()) {
+            $this->path = func_get_arg(1);
+            $paths = 3 <= \func_num_args() ? func_get_arg(2) : [];
+            if (null !== $this->path) {
+                $paths[] = $this->path;
+            }
+
+            if (4 !== \func_num_args() || func_get_arg(3)) {
+                @trigger_error(sprintf('Passing more than one argument to %s is deprecated since Symfony 4.4 and will be removed in 5.0.', __METHOD__), E_USER_DEPRECATED);
+            }
+        } else {
```

Comment:
💡 Avoid the `else` and initialize `$paths = [];` upper

Reply:
I don't agree as it creates an additional assignment and it makes merging changes in master harder where the whole if else get's removed.